lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/scheduler/TaskletScheduler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver.scheduler;
+
+import org.apache.reef.tang.annotations.DefaultImplementation;
+import org.apache.reef.vortex.driver.tasklet.Tasklet;
+import org.apache.reef.vortex.driver.tasklet.TaskletExecutor;
+
+import java.util.Collection;
+
+/**
+ * Chooses an executor to schedule the tasklet to.
+ */
+@DefaultImplementation(RandomTaskletScheduler.class)
+public interface TaskletScheduler {
+  /**
+   * Choose a TaskletExecutor to run a Tasklet.
+   * One can configure and experiment with different scheduling policies
+   * by implementing different flavors of this method.
+   */
+  TaskletExecutor schedule(final Collection<TaskletExecutor> availableExecutors, final Tasklet tasklet);
```

Comment:
Do you have to pass availableExecutors? 
Do you think we can pass it when we construct a `TaskletScheduler` instance? 
Do you expect to have a set of `availableExecutors`?